### DISCUSSION TITLE
An option for output filenames to represent the settings that were used to generate the output.

### DISCUSSION THREAD

**Discussion by beosliege** [nagadomi/nunif#126] [Ideas]
I like to generate several small output files and then view them in my headset to decide which settings I will use for a project. I often find myself renaming the files manually to remember which settings were used for each file. Would it be possible to have an option to place the major settings as part of the filename? E.G. the 3d strength, convergence, method, depth model, resolution, foreground, and edge fix?

If the current input file is: sample.mp4
iw3 will produce an output filename like: sample_LRF_Full_SBS.mp4

Possible new naming convention that would be optional if the user enables the additional values as part of the filename:
sample_3D15_C05_forward_fill_Any_L_DR512_F2_E3_FR00_LRF_Full_SBS.mp4

The above would mean: 
3D15 - 3D Strength was set to 1.5
C05 - Convergence Plane was set to 0.5
forward_fill - the Method used
Any_L - The Depth Model used
DR512 - Depth Resolution was set to 512
F2 - Foreground Scale was set to 2
E3 - Edge Fix was set to 3
FR00 - Flicker Reduction was set to 0
 
Thanks!

**Comment by nagadomi** [comment]
I will add this option after 2024-06-10

**Comment by diverswan9** [comment]
Cheers!

**Comment by nagadomi** [comment]
I added `Add metadata to filename` option (--metadata in cli).
format
https://github.com/nagadomi/nunif/blob/38f935ba992654811be177df41e9071dad25d5cd/iw3/utils.py#L270-L284

Note that Windows has limitations on file path length (perhaps 256), so you may get an error if the original filename is too long.

**Reply by diverswan9** [reply]
Thanks! Looks good...
